dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2625,190 @@ void FileSprayer::pushParts()
     performTransfer();
 }
 
+static StringBuffer & buildAzureStorageURIBase(IStoragePlane * storagePlane, StringBuffer & uri)
+{
+    const char *accountName = storagePlane->queryStorageApiAccount();
+    uri.appendf("https://%s", accountName);
+
+    if (storagePlane->getStorageType()==StorageType::StorageTypeAzureFile)
+        uri.append(".file");
+    else
+        uri.append(".blob");
+    uri.appendf(".core.windows.net/%s", storagePlane->queryStorageContainer());
+
+    return uri;
+}
+
+using namespace Azure::Storage;
+using namespace Azure::Storage::Files;
+using namespace Azure::Storage::Blobs;
+
+bool FileSprayer::transferUsingAPI()
+{
+    // N.B. ensure api copy is possible by making sure useAPICopy() has returned true
+    // TODO:
+    // 1) bool needCalcCRC = calcCRC(); // How should CRC be handled?
+    // 2) file renames
+    // 3) allow access using "AZURE_ACCOUNT_KEY" env variable
+    // 4) allow access using connection string
+    // 5) support striping with multiple containers/file shares
+    // 6) handle overwrite flag
+    // 7) authenticate using managed identity
+
+    // Source
+    StringBuffer sourceClusterName, sourceBaseURI, sourceSAStoken;
+    distributedSource->getClusterName(0, sourceClusterName);
+    Owned<IStoragePlane> sourcePlane = getDataStoragePlane(sourceClusterName.str(), false);
+    buildAzureStorageURIBase(sourcePlane, sourceBaseURI);
+    sourcePlane->querySASToken(sourceSAStoken);
+    const char * sourcePrefix = sourcePlane->queryPrefix();
+    unsigned sourcePrefixLength = isEmptyString(sourcePrefix)?0:strlen(sourcePrefix);
+
+    // Target
+    StringBuffer targetClusterName, targetBaseURI, targetSAStoken;
+    distributedTarget->getClusterName(0, targetClusterName);
+    Owned<IStoragePlane> targetPlane = getDataStoragePlane(targetClusterName.str(), false);
+    buildAzureStorageURIBase(targetPlane, targetBaseURI);
+    targetPlane->querySASToken(targetSAStoken);
+    const char * targetPrefix = targetPlane->queryPrefix();
+    unsigned targetPrefixLength = isEmptyString(targetPrefix)?0:strlen(targetPrefix);
+    auto targetType = targetPlane->getStorageType();
```

Comment:
style: restrict use of auto to cases where the type is awful/impossible to determine.